test/e2e/common/pods.go

```diff
@@ -771,8 +771,13 @@ var _ = framework.KubeDescribe("Pods", func() {
 		}
 	})
 
+	/*
+		Release : v1.15
+		Testname: Ensure readiness gates are respected
+		Description: Create a pod with multiple readiness gates and ensure the pod is unReady until all gates are ready
+	*/
 	// TODO(freehan): label the test to be [NodeConformance] after tests are proven to be stable.
-	ginkgo.It("should support pod readiness gates [NodeFeature:PodReadinessGate]", func() {
+	framework.ConformanceIt("should support pod readiness gates [NodeFeature:PodReadinessGate]", func() {
```

Comment:
I think you should use the agnhost image with the pause command.

Reply:
@bobymcbobs something like this:

```go
Spec: v1.PodSpec{
  TerminationGracePeriodSeconds: &immediate,
    Containers: []v1.Container{
{
  Name:  "agnhost-pause",
  Image: imageutils.GetE2EImage(imageutils.Agnhost),
  Args:  []string{"pause"},
 },
```

Reply:
@hh and @johnbelamaric, I've swapped out the busybox image for Agnhost.

Reply:
I can't bear having a "sleep" in a test like this. It just screams "one day I will become flaky". Can we change the sleep to blocking on a watch with a 20 second timeout or something similar?

Reply:
Do we have a recommended way to perform watches within conformance tests?
I've seen several approaches, and might be nice to have a preferred one pointed out.

Do we want to create the re-usable wait function ala:
https://github.com/kubernetes/kubernetes/pull/79667/files#diff-b017c204d2b1afdc6ed2c313afaffcdcR460

Or should we write some small code wrapping a wait.